tdp/cli/queries.py

```diff
@@ -217,15 +217,17 @@ def get_operation_records(
         List of matching operation records.
 
     Raises:
-        NoResultFound: If the operation does not exist.
+        NoResultFound: If the operation or deployment does not exist.
     """
-    try:
-        return (
-            session.query(OperationModel)
-            .filter_by(deployment_id=deployment_id, operation=operation_name)
-            .all()
-        )
-    except NoResultFound as e:
+    query_result = (
+        session.query(OperationModel)
+        .filter_by(deployment_id=deployment_id, operation=operation_name)
+        .all()
+    )
+    if query_result:
+        return query_result
+
+    else:
         raise Exception(
-            f"Operation {operation_name} does not exist in deployment {deployment_id}."
-        ) from e
+            f"Operation {operation_name} does not exist in deployment {deployment_id} or deployment {deployment_id} does not exist."
+        )
```

Comment:
You can even be more generic:

```suggestion
            f"Can't find operation {operation_name} for deployment {deployment_id}."
        ) from e
```

Reply:
The whole point of the prolonged message was to state the two possible issues with the query, either the operation does not exist in the chosen deployment or the deployment itself does not exist. this message is more specific.
However I can change above the `raise Exception` to  `raise NoResultsFound` which is also more specific.